tomviz/SaveLoadTemplateReaction.cxx

```diff
@@ -0,0 +1,121 @@
+/* This source file is part of the Tomviz project, https://tomviz.org/.
+   It is released under the 3-Clause BSD License, see "LICENSE". */
+
+#include "SaveLoadTemplateReaction.h"
+
+#include "ActiveObjects.h"
+#include "ModuleManager.h"
+#include "Pipeline.h"
+#include "RecentFilesMenu.h"
+#include "Utilities.h"
+
+#include <QApplication>
+#include <QDebug>
+#include <QInputDialog>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QString>
+
+namespace tomviz {
+
+SaveLoadTemplateReaction::SaveLoadTemplateReaction(QAction* parentObject, bool load, QString filename)
+  : pqReaction(parentObject), m_load(load), m_filename(filename)
+{}
+
+void SaveLoadTemplateReaction::onTriggered()
+{
+  if (m_load) {
+    loadTemplate(m_filename);
+  } else {
+    saveTemplate();
+  }
+}
+
+bool SaveLoadTemplateReaction::saveTemplate()
+{
+  bool ok;
+  QString text = QInputDialog::getText(tomviz::mainWidget(), tr("Save Pipeline Template"),
+                                         tr("Template Name:"), QLineEdit::Normal, QString(), &ok);
+  QString fileName = text.replace(" ", "_");
+  if (ok && !text.isEmpty()) {
+    QString path = QApplication::applicationDirPath() +
+                    "/../share/tomviz/templates/" + fileName + ".tvsm";
+    return SaveLoadTemplateReaction::saveTemplate(path);
+  }
+  return false;
+}
+
+bool SaveLoadTemplateReaction::loadTemplate(const QString& fileName)
+{
+  QString path = QApplication::applicationDirPath() +
+                 "/../share/tomviz/templates/" + fileName + ".tvsm";
+  QFile openFile(path);
+  openFile.open(QIODevice::ReadOnly);
```

Comment:
We should check that we where successful in opening the file

Reply:
Thank you, fixed.